emailproxy.py

```diff
@@ -337,9 +344,157 @@ def add_account(username):
+        # create a dict of AWS Secret IDs across all accounts
+        return dict.fromkeys(
+            [AppConfig._PARSER.get(account, 'aws_secret') for account in AppConfig._aws_secrets_accounts()], {})
+
+    # noinspection PyGlobalUndefined,PyPackageRequirements
+    @staticmethod
+    def _aws_secrets_boto3_client():
+        try:
+            global boto3
+            import boto3
+            global botocore
+            import botocore.exceptions
+            aws_client = boto3.client('secretsmanager')
+        except Exception as e:
+            Log.error('Failed to load AWS SDK (boto3) - "aws_secret" is specified in config file for at least one',
+                      'account; have you installed dependencies from `requirements-aws-secrets.txt`?')
+            Log.debug('AWS SDK load error:', Log.error_string(e))
+        else:
+            return aws_client
+
+    @staticmethod
+    def _aws_secrets_debug_log_boto3_error(err):
+        Log.debug(err.response['Error']['Code'] + ":",
+                  err.response['Error']['Message'])
+
+    @staticmethod
+    def _aws_secrets_fetch():
+        aws_client = AppConfig._aws_secrets_boto3_client()
+        if aws_client:
+            # get the values of all AWS Secrets referenced in the configuration file
+            aws_secrets = AppConfig._aws_secrets_ids()
+            for secret_id in aws_secrets:
+                try:
+                    get_secret_value_response = aws_client.get_secret_value(SecretId=secret_id)
+
+                except botocore.exceptions.ClientError as err_get_secret:
+                    if err_get_secret.response['Error']['Code'] == 'ResourceNotFoundException':
+                        Log.info('Fetching AWS Secret "%s"' % secret_id, '- secret does not exist or does not contain',
+                                 'a secret value')
+                    elif err_get_secret.response['Error']['Code'] == 'AccessDeniedException':
+                        Log.error('Fetching AWS Secret "%s"' % secret_id, '- access denied: does the IAM user have the',
+                                  '"secretsmanager:GetSecretValue" permission?')
+                    else:
+                        Log.error('Fetching AWS Secret "%s"' % secret_id, '- unexpected error; see the proxy\'s debug',
+                                  'mode log for details')
+                    AppConfig._aws_secrets_debug_log_boto3_error(err_get_secret)
+
+                else:
+                    aws_secrets[secret_id] = json.loads(get_secret_value_response['SecretString'])
+
+            # update local proxy configuration in memory
+            for account in AppConfig._aws_secrets_accounts():
+                account_secret = AppConfig._PARSER.get(account, 'aws_secret')
+                if account in aws_secrets[account_secret]:
+                    for key in AppConfig._SECRETS_MANAGER_KEYS:
+                        AppConfig._PARSER.set(account, key, aws_secrets[account_secret][account][key])
```

Comment:
```suggestion
                    for key in [k for k in AppConfig._SECRETS_MANAGER_KEYS if k in aws_secrets[account_secret][account]]:
                        AppConfig._PARSER.set(account, key, aws_secrets[account_secret][account].get(key))
```

Here's a potentially preferable solution, which avoids the use of an empty string to denote an undefined key.

Reply:
```suggestion
                    for key in AppConfig._SECRETS_MANAGER_KEYS:
                        if k in aws_secrets[account_secret][account]:
                            AppConfig._PARSER.set(account, key, aws_secrets[account_secret][account][key])
```

I think for future legibility this is a simpler approach.

Reply:
Agreed your proposal is cleaner, added to commit e11338d